src/ProductConstructionService/ProductConstructionService.Api/Api/v2020_02_20/Controllers/PullRequestController.cs

```diff
@@ -111,18 +113,73 @@ public async Task<IActionResult> GetTrackedPullRequests()
             prs.Add(new TrackedPullRequest(
                 key.Replace(keyPrefix, null, StringComparison.InvariantCultureIgnoreCase),
                 TurnApiUrlToWebsite(pr.Url, org, repoName),
-                sampleSub?.Channel != null ? new Channel(sampleSub?.Channel) : null,
+                sampleSub?.Channel != null ? new Channel(sampleSub.Channel) : null,
                 sampleSub?.TargetBranch,
                 sampleSub?.SourceEnabled ?? false,
                 pr.LastUpdate,
                 pr.LastCheck,
                 pr.NextCheck,
-                updates));
+                updates,
+                pr.MergeState));
         }
 
         return Ok(prs.AsQueryable());
     }
 
+    /// <summary>
+    /// Retrieves the tracked codeflow pull request associated with the specified subscription identifier.
+    /// </summary>
+    /// <param name="subscriptionId">The id of the subscription for which to retrieve the tracked pull request.</param>
+    /// <returns>An <see cref="IActionResult"/> containing the tracked pull request details if found; otherwise, a <see
+    /// cref="NotFoundResult"/> if no pull request is associated with the specified subscription.</returns>
+    [HttpGet("tracked/subscription/{subscriptionId}")]
+    [SwaggerApiResponse(HttpStatusCode.OK,
+        Type = typeof(TrackedPullRequest),
+        Description = "Get a PR that is tracked by the service, by its subscription id")]
+    [ValidateModelState]
+    public async Task<IActionResult> GetTrackedPullRequestBySubscriptionId(string subscriptionId)
+    {
+        if (!Guid.TryParse(subscriptionId, out var subscriptionGuid))
+        {
+            return BadRequest("A valid subscription GUID must be provided.");
+        }
+
+        var subscription = await _sqlClient.GetSubscriptionDAOAsync(subscriptionGuid);
```

Comment:
Rest of the controllers use the `_context` directly. I think we should do it here too and then we'd not need the `DAO` methods on the SQL interface?
To me it sounds like an extra layer of indirection that's not really needed?